Panel refactor rows

msecret opened the pull request:
Refactors the rows by creating new components to replace them. I currently haven't refactored the complex usage and limits UI, that might need another component.

Commit:
Allow actions to be placed side by side

By ensuring they have spacing between them

Commit:
Try and refactoring rows

This uses two new components; complex lists and elastic lines. Complex
lists is a way to show list data in a bordered and nested package.
Elastic line can display data in a line with flexible properties on
how the data is displayed.

adborden commented on src/css/components/elastic_line.scss:
```diff
@@ -0,0 +1,20 @@
+@import '../core.scss';
+@import '../../../node_modules/uswds/src/stylesheets/core/defaults.scss';
+
+.elastic_line {
```
good name :)

adborden commented on src/css/components/panel-dep.scss:
```diff
@@ -0,0 +1,168 @@
+  }
+}
+
+.panel-row-bordered {
```
Can we add an example of this and `-boxed` to the pattern library examples?

msecret replied:
I'm going to get rid of this file, but will have examples of all new stuff

adborden reviewed (commented):
:+1:

Commit:
Brought complex lists and elastic rows into dash

And made some changes ehre based on what I found there. I found that
it was easier in react to not nest the complex list in a list item but
to just put it by it's side. Also changed how borders were displayed
to allow this.

Commit:
Add functionality for empty compelx list

To ensure it renders with the single line above it as a full
list.

Commit:
Expandable box component

And fixes to pattern library with all the panel row stuff being
taken out.

Expandable box is a component to take care of the overview page
quicklooks where you can click on one to expand it. I also changed
card the card and panel features to more specific to actual cloud.gov.

adborden commented on src/css/components/expandable_box.scss:
```diff
@@ -0,0 +1,26 @@
+$box-padding: $grid-3;
+
+.expandable_box {
+  background-color: white;
```
Is this necessary? We can't just use transparent?
